src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -249,6 +258,15 @@ def load_arguments(self, _):
         c.argument('handle_extended_json_format', arg_type=extended_json_format_type,
                    deprecate_info=c.deprecate(target='--handle-extended-json-format/-j'))
         c.argument('no_prompt', arg_type=no_prompt)
+        c.argument('confirm_with_what_if', arg_type=deployment_what_if_confirmation_type, is_preview=True)
+        c.argument('what_if_result_format', options_list=['--what-if-result-format', '-r'],
+                   arg_type=deployment_what_if_result_format_type, is_preview=True)
+
+    with self.argument_context('deployment sub what-if') as c:
```

Comment:
Does the `what if` command of sub need to support `aux_tenants`?

Reply:
No. If I understand it correctly, the parameter is only useful for cross tenant resource group vm deployment: https://github.com/Azure/azure-cli/issues/11151.